staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -259,6 +260,14 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			if len(rule.FieldPath) != 0 {
+				validFieldPath, err := ValidFieldPath(rule.FieldPath, fldPath, sts)
```

Comment:
this looks like we're doing this parsing/validation on every error we return on every CR validation... am I reading that correctly? we should only do this once at CRD load time, and transform valid field paths to stored normalized string field paths we can just append to the parent fldPath, so this would become something like this at validation time:

```go
if normalizedRuleFieldPath := s.normalizedRuleFieldPaths[i]; len(normalizedRuleFieldPath) > 0 {
  fldPath = fldPath.Child(normalizedRuleFieldPath)
}
```

I also don't think we should fail CR validation if there was a field path parsing error at CRD load time, and just fall back to using the unqualified fldPath

Reply:
Have moved it to compilation so it will only be called once.